include/h2o/tracing.h

```diff
@@ -0,0 +1,42 @@
+#ifndef h2o__tracing_h
+#define h2o__tracing_h
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// MSB indicates tracepoint should be disabled
+#define H2O_TRACE_REMOVE_PROBE ((uint64_t)1<<63)
+
+// list of available tracepoints
+#define H2O_TRACE_SSLNEW     ((uint64_t)1<<0)
+#define H2O_TRACE_H1NEW      ((uint64_t)1<<1)
+#define H2O_TRACE_H2NEW      ((uint64_t)1<<2)
+#define H2O_TRACE_REQ        ((uint64_t)1<<3)
+#define H2O_TRACE_RES        ((uint64_t)1<<4)
+#define H2O_TRACE_TXHD       ((uint64_t)1<<5)
+#define H2O_TRACE_RXHD       ((uint64_t)1<<6)
+#define H2O_TRACE_PROXY_REQ  ((uint64_t)1<<7)
+#define H2O_TRACE_PROXY_RES  ((uint64_t)1<<8)
+#define H2O_TRACE_PROXY_TXHD ((uint64_t)1<<9)
+#define H2O_TRACE_PROXY_RXHD ((uint64_t)1<<10)
+
+struct st_h2o_tracing_conn_t {
+    h2o_conn_t super;
```

Comment:
I think you do not need to inherit `h2o_conn_t`. `h2o_conn_t` is an abstraction of an HTTP connection (either H1, H2, H3). It exists to help logging accesses to such connections and providing hooks for HTTP version-specific APIs.

Assuming that we are not implementing our own version of HTTP protocol here, what you can do is simply:
* in src/main.c, create a listening socket and accept tracing connections
* pass the established connection to `h2o_tracing_accept`